partner_address_version/models/res_partner.py

```diff
@@ -13,54 +13,69 @@ class ResPartner(models.Model):
     version_hash = fields.Char(readonly=True, copy=False)
     date_version = fields.Datetime(string='Date version', readonly=True)
 
-    @api.multi
-    def get_address_version(self):
-        """
-        Get a versioned partner corresponding to the partner fields.
-        If no versioned partner exists, create a new one.
-        """
-        self.ensure_one()
-        version_hash = self.get_version_hash()
-        versioned_partner = self.with_context(active_test=False).search(
-            [('version_hash', '=', version_hash)])
-        if versioned_partner:
-            return versioned_partner
-        default = {
-            'active': False,
-            'version_hash': version_hash,
-            'parent_id': self.parent_id and self.parent_id.id or self.id,
-            'date_version': fields.Datetime.now(),
-        }
-        versioned_partner = self.copy(default=default)
-        return versioned_partner
-
-    def get_version_fields(self):
+    def _version_fields(self):
         return [
             'name', 'street', 'street2', 'zip', 'city', 'country_id',
             'parent_id'
         ]
 
-    def get_version_hash(self):
-        version_fields = self.get_version_fields()
+    def _version_hash(self):
+        version_fields = self._version_fields()
         version = OrderedDict()
         for field in version_fields:
             if field == 'parent_id':
-                parent_id = self.parent_id and self.parent_id.id or self.id
+                parent_id = self.parent_id.id if self.parent_id else self.id
                 version[field] = parent_id
             elif self[field]:
                 version[field] = self[field]
         version_hash = hashlib.md5(str(version)).hexdigest()
         return version_hash
 
+    def _version_impacted_tables(self):
+        """
+        :return:
+            - list of tables to update in case of address versioning
+        """
+        return []
+
+    def _version_need(self):
```

Comment:
hmm this is wanted? `_version_apply` won't run...

Reply:
this is supposed to be overriden.
See https://github.com/OCA/sale-workflow/pull/1119/files#diff-9631fa4efa5849317b2d7720e17c8a75R11

Reply:
then please add a docstring ;)

Reply:
indeed, done ;-)